File: lucybell.client/src/app/Models/Producto.ts
import {ImagenProductoDTO} from "../Models/ImagenProducto"
import {VariantesProducto} from "../Models/VariantesProducto"
export interface Producto {
  id: number;
  nombre: string;
  precio: number;
  descripcion?: string;
  destacado: boolean;
  categoriaId: number;
  categoriaNombre: string;
  subCategoriaId: number;
  materialId: number;
  imagenesProductos: ImagenProductoDTO[];
  variantesProducto: VariantesProducto[];
  
}

export interface PaginatedProductos {
  productos: Producto[];
  totalCount: number;
  totalPages: number;
}

export interface ProductoCreacion {
  nombre: string;
  precio: number;
  destacado: boolean;
  descripcion?: string;
}

export interface ProductoSinVariantesDTO {
  id: number;
  nombre: string;
  precio: number;
  descripcion?: string;
}

File: LucyBell.Server/Entidades/Producto.cs
﻿using System.ComponentModel.DataAnnotations;

namespace LucyBell.Server.Entidades
{
	public class Producto
	{
		public int Id { get; set; }

		[Required(ErrorMessage = "El campo {0} es requerido")]
		[StringLength(maximumLength: 120, ErrorMessage = "El campo {0} no debe tener más de {1} carácteres")]
		public string Nombre { get; set; }

		[Required(ErrorMessage = "El campo {0} es requerido")]
		public decimal Precio { get; set; }

		public string? Descripcion { get; set; }
		public bool Destacado { get; set; }

		public List<VarianteProducto> VariantesProducto { get; set; }
        public List<ImagenProducto> ImagenesProductos { get; set; }
        public int CategoriaId { get; set; }
		public Categoria Categoria { get; set; }
		public int? SubCategoriaId { get; set; }
		public SubCategoria SubCategoria { get; set; }
		public int? MaterialId { get; set; }
		public Material Material { get; set; }
	}
}
